packages/ts/src/components/treemap/index.ts

```diff
@@ -0,0 +1,307 @@
+import { Selection, select } from 'd3-selection'
+import { hierarchy, HierarchyNode, treemap } from 'd3-hierarchy'
+import { group, max, extent } from 'd3-array'
+import { scaleLinear, scaleThreshold } from 'd3-scale'
+import { hsl } from 'd3-color'
+import { wrapSVGText } from 'utils/text'
+import { ComponentCore } from 'core/component'
+import { SeriesDataModel } from 'data-models/series'
+import { getColor, brighter, getHexValue, isDarkBackground } from 'utils/color'
+import { getString, getNumber } from 'utils/data'
+import { TreemapConfigInterface, TreemapDefaultConfig } from './config'
+import { TreemapDatum, TreemapNode } from './types'
+
+import * as s from './style' // Minimum pixel size for showing labels
+
+const LIGHTNESS_VARIATION_AMOUNT = 0.08
+
+const MIN_TILE_SIZE_FOR_LABEL = 20
+
+const defaultNumberFormat = (value: number): string => `${value}`
+
+export class Treemap<Datum> extends ComponentCore<Datum[], TreemapConfigInterface<Datum>> {
+  static selectors = s
+  protected _defaultConfig = TreemapDefaultConfig as TreemapConfigInterface<Datum>
+  public config: TreemapConfigInterface<Datum> = this._defaultConfig
+
+  datamodel: SeriesDataModel<Datum> = new SeriesDataModel()
+  tiles: Selection<SVGGElement, unknown, SVGGElement, unknown>
+
+  private isTileLargeEnough (d: TreemapNode<Datum>): boolean {
```

Comment:
I've started prefixing all private variables and functions with `_` but haven't added an ESLint rule for that yet. While it looks slightly less clean, I think it improves core readability and I would like us to keep doing that.